libcaf_net/caf/net/datagram_transport.hpp

```diff
@@ -0,0 +1,327 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2019 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include <deque>
+#include <unordered_map>
+
+#include "caf/byte.hpp"
+#include "caf/detail/socket_sys_aliases.hpp"
+#include "caf/detail/socket_sys_includes.hpp"
```

Comment:
These header should never get pulled into a CAF header. Especially `socket_sys_includes`, which pulls in Windows headers that can mess up downstream code. Why did you need them here? If something is missing from the interface for `udp_datagram_socket`?

Reply:
The forward declaration of `udp_datagram_socket` was missing in `net::fwd.hpp`. I fixed that issue already in the quic-branch.

Reply:
These must have slipped into the file somehow.. I don't need them.